api/src/test/java/org/openmrs/api/db/hibernate/HibernateFormDAOTest.java

```diff
@@ -0,0 +1,448 @@
+		
+		String partialName = "";
+		Boolean published = null;
+		Collection<EncounterType> encounterTypes = new HashSet<EncounterType>();
+		encounterTypes.add(encounterService.getEncounterType(3));
+		Boolean retired = null;
+		Collection<FormField> containingAnyFormField = new HashSet<FormField>();
+		Collection<FormField> containingAllFormFields = new HashSet<FormField>();
+		Collection<Field> fields = new HashSet<Field>();
+		
+		List<Form> formsActual = hibernateFormDAO.getForms(partialName, published, encounterTypes, retired,
+		    containingAnyFormField, containingAllFormFields, fields);
+		Integer formCountActual = hibernateFormDAO.getFormCount(partialName, published, encounterTypes, retired,
+		    containingAnyFormField, containingAllFormFields, fields);
+		
+		ArrayList<Form> expectedForms = new ArrayList<Form>();
+		
+		Assert.assertEquals(expectedForms, formsActual);
+		Assert.assertEquals(Integer.valueOf(0), formCountActual);
+	}
+	
+	@Test
+	public void shouldReturn1FormWhenGivenRetiredIsFalse() {
+		FormService formService = Context.getFormService();
+		
+		String partialName = "";
+		Boolean published = null;
+		Collection<EncounterType> encounterTypes = new HashSet<EncounterType>();
+		Boolean retired = Boolean.FALSE;
+		Collection<FormField> containingAnyFormField = new HashSet<FormField>();
+		Collection<FormField> containingAllFormFields = new HashSet<FormField>();
+		Collection<Field> fields = new HashSet<Field>();
+		
+		List<Form> formsActual = hibernateFormDAO.getForms(partialName, published, encounterTypes, retired,
+		    containingAnyFormField, containingAllFormFields, fields);
+		Integer formCountActual = hibernateFormDAO.getFormCount(partialName, published, encounterTypes, retired,
+		    containingAnyFormField, containingAllFormFields, fields);
+		
+		ArrayList<Form> expectedForms = new ArrayList<Form>();
+		expectedForms.add(formService.getForm(1));
+		
+		Assert.assertEquals(expectedForms, formsActual);
+		Assert.assertEquals(Integer.valueOf(1), formCountActual);
+	}
+	
+	@Test
+	public void shouldReturn2FormsWhenGivenRetiredIsTrue() {
+		FormService formService = Context.getFormService();
+		
+		String partialName = "";
+		Boolean published = null;
+		Collection<EncounterType> encounterTypes = new HashSet<EncounterType>();
+		Boolean retired = Boolean.TRUE;
+		Collection<FormField> containingAnyFormField = new HashSet<FormField>();
+		Collection<FormField> containingAllFormFields = new HashSet<FormField>();
+		Collection<Field> fields = new HashSet<Field>();
+		
+		List<Form> formsActual = hibernateFormDAO.getForms(partialName, published, encounterTypes, retired,
+		    containingAnyFormField, containingAllFormFields, fields);
+		Integer formCountActual = hibernateFormDAO.getFormCount(partialName, published, encounterTypes, retired,
+		    containingAnyFormField, containingAllFormFields, fields);
+		
+		ArrayList<Form> expectedForms = new ArrayList<Form>();
+		expectedForms.add(formService.getForm(2));
```

Comment:
Setting include retired to true should return the same forms as those when you set retired to false with the addition of retired ones. So I'd think this should include the form with id 1 since it got returned in the previous test

Reply:
Regarding the actual functionality of this boolean, the tests we wrote already pass. That means that if the functionality you mention is correct, then there is a bug. We can of course modify tests to show that this is a bug, but I want to confirm it is instead of a reading mistake. If it is a bug, I do want to suggest refactoring the 'retired' variable name to 'includeRetiered' to improve readability.